### PR TITLE
fix: address Centaur review findings from merged PRs

### PR DESCRIPTION
**Pull request by dimakis**
## Summary

Addresses Centaur review comments that were not resolved before merging PRs #273, #277, #279, #281, #283, #284. PR #277 was clean (LGTM).

### SessionSearchBar (#273) — 2 bugs + 3 improvements
- **Bug**: stale results from prior query shown while new search is in progress — gate `results.map()` on `!searching`
- **Bug**: `useState(active)` never syncs after mount — add `useEffect` with `prevActiveRef` to open bar on `false→true` transition
- Add Escape key to dismiss search bar
- Use viewport-safe `max()` calc for results dropdown instead of magic `-60px`
- 4 new tests: stale results hidden, Escape dismisses, active prop sync, close reverts UI

### Store progress reset (#281) — 1 missing test
- Add `progress` reset assertions to both `switchSession` and `newSession` describe blocks

### Design docs (#283) — 9 findings
- Update status: Phase 1 shipped (PR #278), not "pending implementation"
- Mark P3 (generateWtId) and P5 (registry.remove) as fixed in Phase 1
- Remove dead canvas path refs and ephemeral local-machine paths
- Mark `denyPendingBySession` as already implemented in harness
- Mark `active_elsewhere` F6 as no-op (case doesn't exist in parser)
- Annotate deferred items with current relevance

### Phase 2 handoff (#279) — 7 findings
- Fix `generateWtId` description: `randomUUID().replace(/-/g, '').slice(0, 12)`
- Correct 2e resume claim: primary is preserved, secondaries are the problem
- Add failure path for on-demand creation (hard deny, no redirect loop)
- Add double-cleanup note (catch + finally both call cleanup)
- Add missing symlink edge case tests to plan
- Replace stale line references with search instructions
- Remove hardcoded ngrok URL and webhook ID

### Infrastructure (#284) — 4 findings
- Make podman path portable (`command -v` fallback)
- Add `podman machine init` fallback in plist and deploy script
- Remove undocumented `LaunchOnlyOnce` launchd key

### Testing infrastructure
- Exclude `.claude/worktrees/` and `.cursor/worktrees/` from vitest to prevent stale test copies from running

## Test plan

- [x] SessionSearchBar: 13 tests pass (4 new)
- [x] Store: 57 tests pass (2 new)
- [x] Pre-commit hooks pass (lint, format, gitleaks)
- [x] Pre-existing test failures unchanged (TabBar, TaskNode, worktree, token-update, prompt-compare — all unrelated)

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: vitest.config.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ export default defineConfig({
     },
   },
   test: {
+    exclude: ['**/node_modules/**', '**/.claude/worktrees/**', '**/.cursor/worktrees/**'],
```
Comment (dimakis) [inline vitest.config.ts]: 🟡 **bugs**: Setting `exclude` explicitly replaces Vitest's defaults (`**/dist/**`, `**/.{idea,git,cache,output,temp}/**`, `**/cypress/**`, etc.). Tests could accidentally pick up files from `dist/` or `.git/`. Use `import { configDefaults } from 'vitest/config'` and spread: `exclude: [...configDefaults.exclude, '**/.claude/worktrees/**', '**/.cursor/worktrees/**']`. `[fixable]`

File: scripts/deploy.sh [hunk 2]
```diff
@@ -21,8 +21,12 @@ sed "s|__MITZO_HOME__|${MITZO_HOME}|g" infra/com.mitzo.podman-machine.plist > "$
 launchctl bootout "gui/$(id -u)/com.mitzo.podman-machine" 2>/dev/null || true
 launchctl bootstrap "gui/$(id -u)" "$PODMAN_PLIST_DEST"
 
-# Ensure podman machine is running before bringing up containers
+# Ensure podman machine exists and is running before bringing up containers
 if ! podman machine inspect --format '{{.State}}' 2>/dev/null | grep -q Running; then
+  if ! podman machine inspect 2>/dev/null >/dev/null; then
```
Comment (dimakis) [inline scripts/deploy.sh:26]: 🔵 **unsafe_assumptions**: There's a potential race between the launchd plist (which runs `podman machine init && start` at boot via `RunAtLoad`) and this deploy script. If deploy runs shortly after boot, both could try to init/start simultaneously. Low risk in practice since podman uses lock files, but worth a comment noting the plist handles startup.

File: docs/design/session-isolation-phase2-handoff.md [hunk 3]
```diff
@@ -154,19 +160,19 @@ Recommended sequence (dependencies noted):
...
-| `server/constants.ts`                        | `WORKTREE_STALE_HOURS`        | L29 (currently 96)                    |
-| `server/repo-config.ts`                      | `RepoConfig` interface        | L30-36 (`repos` field)                |
-| `packages/harness/src/worktree-guard.ts`     | `checkWorktreePolicy()`       | L80-143 (sync)                        |
-| `packages/harness/src/permission-handler.ts` | `buildPermissionHandler()`    | L44-146 (async `canUseTool`)          |
+| File                                         | What                          | Key lines                                                 |
+| -------------------------------------------- | ----------------------------- | --------------------------------------------------------- |
+| `server/chat.ts`                             | `createSessionWorktrees()`    | L278-338 (secondary loop at L317-332)                     |
+| `server/chat.ts`                             | `buildWorktreeSystemPrompt()` | search for function name (line numbers shift between PRs) |
+| `server/chat.ts`                             | `startChat()` finally block   | search for `} finally {`                                  |
```
Comment (dimakis) [inline docs/design/session-isolation-phase2-handoff.md]: 🔵 **style**: The 'Key lines' column now says 'search for function name' for every row, making the column redundant. Consider either dropping the column entirely or adding more useful hints (e.g., approximate location like 'near L160' or 'in the worktree section'). `[fixable]`

**Review by dimakis — COMMENTED** [review]
## Centaur Review

Found **4** issue(s) (1 warning).


### `vitest.config.ts`
_Solid cleanup PR. One real issue: the vitest `exclude` override drops Vitest's default exclusions (`dist/`, `.git/`, etc.) — use `configDefaults.exclude` spread instead. The rest are minor suggestions._

- 🟡 **bugs** (L17): Setting `exclude` explicitly replaces Vitest's defaults (`**/dist/**`, `**/.{idea,git,cache,output,temp}/**`, `**/cypress/**`, etc.). Tests could accidentally pick up files from `dist/` or `.git/`. Use `import { configDefaults } from 'vitest/config'` and spread: `exclude: [...configDefaults.exclude, '**/.claude/worktrees/**', '**/.cursor/worktrees/**']`. `[fixable]`


### `infra/com.mitzo.podman-machine.plist`
_Solid cleanup PR. One real issue: the vitest `exclude` override drops Vitest's default exclusions (`dist/`, `.git/`, etc.) — use `configDefaults.exclude` spread instead. The rest are minor suggestions._

- 🔵 **unsafe_assumptions**: The bash one-liner `podman machine init 2>/dev/null` silences all init errors (disk full, permission denied, corrupted VM image), not just 'already initialized'. Consider checking `podman machine inspect` first and only running init if it fails, similar to what `deploy.sh` already does. `[fixable]`


### `scripts/deploy.sh`
_Solid cleanup PR. One real issue: the vitest `exclude` override drops Vitest's default exclusions (`dist/`, `.git/`, etc.) — use `configDefaults.exclude` spread instead. The rest are minor suggestions._

- 🔵 **unsafe_assumptions** (L26): There's a potential race between the launchd plist (which runs `podman machine init && start` at boot via `RunAtLoad`) and this deploy script. If deploy runs shortly after boot, both could try to init/start simultaneously. Low risk in practice since podman uses lock files, but worth a comment noting the plist handles startup.


### `docs/design/session-isolation-phase2-handoff.md`
_Solid cleanup PR. One real issue: the vitest `exclude` override drops Vitest's default exclusions (`dist/`, `.git/`, etc.) — use `configDefaults.exclude` spread instead. The rest are minor suggestions._

- 🔵 **style** (L167): The 'Key lines' column now says 'search for function name' for every row, making the column redundant. Consider either dropping the column entirely or adding more useful hints (e.g., approximate location like 'near L160' or 'in the worktree section'). `[fixable]`

File: docs/design/session-isolation-phase2-handoff.md [hunk 4]
```diff
@@ -154,19 +160,19 @@ Recommended sequence (dependencies noted):
...
-| `server/chat.ts`                             | `cleanupSessionWorktrees()`   | L803+                                 |
-| `server/worktree.ts`                         | `createWorktree()` (sync)     | L46-102                               |
-| `server/worktree.ts`                         | `cleanupStaleWorktrees()`     | L271+                                 |
-| `server/constants.ts`                        | `WORKTREE_STALE_HOURS`        | L29 (currently 96)                    |
-| `server/repo-config.ts`                      | `RepoConfig` interface        | L30-36 (`repos` field)                |
-| `packages/harness/src/worktree-guard.ts`     | `checkWorktreePolicy()`       | L80-143 (sync)                        |
-| `packages/harness/src/permission-handler.ts` | `buildPermissionHandler()`    | L44-146 (async `canUseTool`)          |
+| File                                         | Symbol / Function                               |
+| -------------------------------------------- | ----------------------------------------------- | ------------------------------------- |
```
Comment (dimakis) [inline docs/design/session-isolation-phase2-handoff.md]: 🟡 **style**: Broken markdown table: the separator line defines 3 columns (`| --- | --- | --- |`) but only the first data row has 3 columns (with a stale line-number reference `L278-338`). The remaining rows have 2 columns. This renders as a malformed table in most markdown renderers. Either add the third column consistently or remove the trailing `| --- |` from the separator and the stale line reference from row 1. `[fixable]`

File: scripts/deploy.sh [hunk 5]
```diff
@@ -21,8 +21,14 @@ sed "s|__MITZO_HOME__|${MITZO_HOME}|g" infra/com.mitzo.podman-machine.plist > "$
 launchctl bootout "gui/$(id -u)/com.mitzo.podman-machine" 2>/dev/null || true
 launchctl bootstrap "gui/$(id -u)" "$PODMAN_PLIST_DEST"
 
-# Ensure podman machine is running before bringing up containers
+# Ensure podman machine exists and is running before bringing up containers.
```
Comment (dimakis) [inline scripts/deploy.sh]: 🔵 **style**: The inline comment block (lines 24-26) explaining redundancy with the launchd plist is helpful context but could be trimmed to one line. Minor — not blocking. `[fixable]`

**Review by dimakis — COMMENTED** [review]
## Centaur Review

Found **3** issue(s) (1 warning).


### `docs/design/session-isolation-phase2-handoff.md`
_Clean follow-up PR. Doc updates accurately reflect shipped state, frontend fixes are well-tested, vitest/podman changes are correct. One broken markdown table to fix; the rest is solid._

- 🟡 **style** (L164): Broken markdown table: the separator line defines 3 columns (`| --- | --- | --- |`) but only the first data row has 3 columns (with a stale line-number reference `L278-338`). The remaining rows have 2 columns. This renders as a malformed table in most markdown renderers. Either add the third column consistently or remove the trailing `| --- |` from the separator and the stale line reference from row 1. `[fixable]`


### `infra/com.mitzo.podman-machine.plist`
_Clean follow-up PR. Doc updates accurately reflect shipped state, frontend fixes are well-tested, vitest/podman changes are correct. One broken markdown table to fix; the rest is solid._

- 🔵 **missing_tests**: The plist now runs `podman machine init 2>/dev/null` unconditionally (stderr suppressed), then `exec podman machine start`. If `machine init` fails for a reason other than 'already exists' (e.g. disk full, permission issue), the error is silently swallowed and `machine start` will fail with a confusing error. Consider redirecting only the expected 'already exists' exit code rather than blanket `2>/dev/null`, or at minimum logging that init was attempted. This is low-risk since it's a boot-time convenience, but worth noting. `[fixable]`


### `scripts/deploy.sh`
_Clean follow-up PR. Doc updates accurately reflect shipped state, frontend fixes are well-tested, vitest/podman changes are correct. One broken markdown table to fix; the rest is solid._

- 🔵 **style** (L24): The inline comment block (lines 24-26) explaining redundancy with the launchd plist is helpful context but could be trimmed to one line. Minor — not blocking. `[fixable]`
